### ISSUE TITLE
v18.09.0 does not embed version number in ISO header

### ISSUE THREAD

**Issue by peterlloydcc** [boot2docker/boot2docker#1347]
docker-machine looks for a version number in the boot2docker ISO header. See here:
https://github.com/docker/machine/blob/master/libmachine/mcnutils/b2d.go#L310

v18.09.0 does not appear to be embedding a version number in the same way as previously.
Offset 0x8028 in v18.09.0 ISO has the string `Boot2Docker`
Offset 0x8028 in v18.06.1-ce ISO has the string `b2d-v18.06.1-ce`

The consequence of this is that docker-machine repeatedly logs
`Unable to get the local Boot2Docker ISO version:  Did not find prefix \"-v\" in version string"`
`Default Boot2Docker ISO is out-of-date, downloading the latest release...`

And, more seriously, it does re-download the boot2docker ISO for every machine created!

**Comment by daniel-shimon** [comment]
Had to edit add it with HexEditor 🤣 
Just adding "-v18.09.0" to offset 0x8032 solves it of course 👍

**Comment by pavanssetty** [comment]
Why cannot be boot2docker ISO file fixed rather?

**Comment by tianon** [comment]
The fix will be in the next release of boot2docker.

**Comment by rcadhikari** [comment]
Run the following following steps:

1. Check your docker version:
```
$ docker -v
Docker version 18.03.0-ce, build 0520e24
```

2. Download your docker version's of `boot2docker` (18.03.0-ce) from [here][1] into your local:
```
wget https://github.com/boot2docker/boot2docker/releases/download/v18.03.0-ce/boot2docker.iso -P ~/.docker/machine/cache/test/boot2docker-v18-03-0-ce.iso
```

3. Create the `new_virtual_box_name` using the downloaded version of `boot2docker`:
```
docker-machine create --driver virtualbox --virtualbox-boot2docker-url ~/.docker/machine/cache/test/boot2docker-v18-03-0-ce.iso new_virtual_box_name
```

  [1]: https://github.com/boot2docker/boot2docker/releases?after=v18.04.0-ce-rc1

**Comment by tianon** [comment]
See https://github.com/boot2docker/boot2docker/releases/tag/v18.09.1-beta1 and https://github.com/boot2docker/boot2docker/releases/tag/v18.09.1-beta2 (which include this fix). :+1:

**Comment by kickbotowski** [comment]
I used
https://github.com/boot2docker/boot2docker/releases/tag/v18.09.1-beta2 
Although Did not find prefix "-v" not display anymore,  but it still download (default) Default Boot2Docker ISO is out-of-date and re download(Window 7). Is any ideas?

**Comment by chanjamin** [comment]
> Had to edit add it with HexEditor 🤣
> Just adding "-v18.09.0" to offset 0x8032 solves it of course 👍
how to?

**Comment by siweiqi** [comment]
@daniel-shimon   how to insert  "-v18.09.0" ??

**Comment by 141801** [comment]
sh-3.2# docker-machine  create --driver virtualbox  default "-v18.09.0"
Invalid command line. Found extra arguments [-v18.09.0]

**Comment by avoidik** [comment]
```bash
printf '\x2D\x76\x31\x38\x2E\x30\x39\x2E\x30\x20' | dd of=boot2docker.iso bs=1 seek=32819 count=10 conv=notrunc 
```

**Comment by debragail** [comment]
> The fix will be in the next release of boot2docker.

is it fixed?

**Comment by tianon** [comment]
https://github.com/boot2docker/boot2docker/issues/1347#issuecomment-444273380

**Comment by tianon** [comment]
My comment notes that it's fixed in the pre-releases for 18.09 -- there is not a non-pre-release build that includes the fix (and by default, `docker-machine` ignores pre-releases).  If you manually use `--XYZ-boot2docker-url` in order to use one of those pre-release versions, it should note that the version you have is newer than the version available and stop re-downloading every time, or you can hex edit your ISO file, or you can wait another day or two for the official v18.09.1 release.
